std/math.d

```diff
@@ -4103,9 +4103,54 @@ long lrint(real x) @trusted pure nothrow @nogc
 
             return sign ? -result : result;
         }
+        else static if (F.realFormat == RealFormat.ieeeQuadruple)
+        {
+            long result;
+
+            // Rounding limit when casting from real(128-bit) to ulong.
+            enum real OF = 0x1P+112;
+
+            ulong* vl = cast(ulong*)(&x);
+
+            enum EXPSHIFT_ULONG = 48;
+            enum MANTISSE = real.mant_dig - 1;
+
+            // Find the exponent and sign
+            ulong hi = vl[MANTISSA_MSB];
+            ulong lo = vl[MANTISSA_LSB];
+            int exp = cast(int)((hi >> EXPSHIFT_ULONG) & F.EXPMASK) - F.EXPBIAS;
+            int sign = cast(int)(lo >> 63);
```

Comment:
This should be:
```D
ulong hi = vl[MANTISSA_MSB];
int exp = cast(int)((hi >> EXPSHIFT_ULONG) & F.EXPMASK) - F.EXPBIAS;
int sign = cast(int)(hi >> 63);
```
The sign bit is in the `hi` part, which in turn means that you only actually need the `lo` part one place:
```D
else
{
    const lo = vl[MANTISSA_LSB];
    result = (cast(long) hi << (exp - EXPSHIFT_ULONG)) | (lo >> (MANTISSE - exp));
}
```